lib/shopify-cli/tasks/ensure_env.rb

```diff
@@ -3,27 +3,91 @@
 module ShopifyCli
   module Tasks
     class EnsureEnv < ShopifyCli::Task
-      def call(ctx)
+      def call(ctx, regenerate_env: false)
         @ctx = ctx
-        Resources::EnvFile.read(ctx.root)
+        if regenerate_env
+          ask
+        else
+          env = Resources::EnvFile.read(ctx.root)
+          ask unless [:api_key, :secret, :shop].all? { |property| env[property] }
+        end
       rescue Errno::ENOENT
         ask
       end
 
       def ask
-        api_key = CLI::UI.ask(@ctx.message('core.tasks.ensure_env.api_key_question'))
-        api_secret = CLI::UI.ask(@ctx.message('core.tasks.ensure_env.api_secret_key_question'))
-        shop = CLI::UI.ask(@ctx.message('core.tasks.ensure_env.development_store_question'))
-        shop.gsub!(/https?\:\/\//, '')
-
-        env = Resources::EnvFile.new(
-          api_key: api_key,
-          secret: api_secret,
+        env_data = begin
+          Resources::EnvFile.parse_external_env
+                   rescue Errno::ENOENT
+                     {}
+        end
```

Comment:
This here and line 8-12 above have some duplication.  We should have `ask` to ONLY just prompt assuming that by the time it's called, it doesn't need to check if `.env` file exits.  The `ask` should prompt and return the values of the questions asked (to get api_key and etc).  The read/write from/into the .env file should be in the `call`.  In fact, you can see that there are duplicated read from .env file logic on line 11.